vector/src/main/java/im/vector/app/features/location/live/GetLiveLocationShareSummaryUseCase.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live
+
+import androidx.annotation.MainThread
+import androidx.lifecycle.asFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.mapNotNull
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.session.room.model.livelocation.LiveLocationShareAggregatedSummary
+import timber.log.Timber
+import javax.inject.Inject
+
+class GetLiveLocationShareSummaryUseCase @Inject constructor(
+        private val session: Session,
+) {
+
+    @MainThread
+    fun execute(roomId: String, eventId: String): Flow<LiveLocationShareAggregatedSummary> {
```

Comment:
not a blocker, I'm wondering if we could/should stick with coroutines here and rely on a `withContext(Main)` (injecting the dispatchers) :thinking:

Reply:
You are right. When calling a usecase dealing with coroutine, we should not have to worry about the `Dispatcher` in which it should run. I have done some changes here: [7099665](https://github.com/vector-im/element-android/pull/6356/commits/70996655e150dc5f3cb246e0d06e0b7e7fdb2a6d). I also fixed an issue with the `FakeSession` and its dispatchers (see [8476fba](https://github.com/vector-im/element-android/pull/6356/commits/8476fba5ff1d5140e755b9085c3f4b5ad0a1f6c1)).